docs/reference/analysis/tokenfilters/dictionary-decompounder-tokenfilter.asciidoc

```diff
@@ -0,0 +1,173 @@
+=== Dictionary decompounder token filter
+++++
+<titleabbrev>Dictionary decompounder</titleabbrev>
+++++
+
+[NOTE]
+====
+In most cases, we recommend using the faster
+<<analysis-hyp-decomp-tokenfilter,`hyphenation_decompounder`>> token filter
+in place of this filter. However, you can use the
+`dictionary_decompounder` filter to check the quality of a word list before
+implementing it in the `hyphenation_decompounder` filter.
+====
+
+Uses a specified list of words and a brute force approach to find subwords in
+compound words. If found, these subwords are included in the token output.
+
+This filter uses Lucene's
+https://lucene.apache.org/core/{lucene_version_path}/analyzers-common/org/apache/lucene/analysis/compound/DictionaryCompoundWordTokenFilter.html[DictionaryCompoundWordTokenFilter],
+which was built for Germanic languages.
+
+[[analysis-dict-decomp-tokenfilter-analyze-ex]]
+==== Example
+
+The following <<indices-analyze,analyze API>> request uses the
+`dictionary_decompounder` filter to find subwords in `Donaudampfschiff`. The
+filter then checks these subwords against a specified list of words: `Donau`,
+`dampf`, `meer`, and `schiff`.
+
+[source,console]
+--------------------------------------------------
+GET _analyze
+{
+  "tokenizer": "standard",
+  "filter": [
+    {
+      "type": "dictionary_decompounder",
+      "word_list": ["Donau", "dampf", "meer", "schiff"]
+    }
+  ],
+  "text": "Donaudampfschiff"
+}
+--------------------------------------------------
+
+The filter produces the following tokens:
+
+[source,text]
+--------------------------------------------------
+[ Donaudampfschiff, Donau, dampf, schiff ]
+--------------------------------------------------
+
+/////////////////////
+[source,console-result]
+--------------------------------------------------
+{
+  "tokens" : [
+    {
+      "token" : "Donaudampfschiff",
+      "start_offset" : 0,
+      "end_offset" : 16,
+      "type" : "<ALPHANUM>",
+      "position" : 0
+    },
+    {
+      "token" : "Donau",
+      "start_offset" : 0,
+      "end_offset" : 16,
+      "type" : "<ALPHANUM>",
+      "position" : 0
+    },
+    {
+      "token" : "dampf",
+      "start_offset" : 0,
+      "end_offset" : 16,
+      "type" : "<ALPHANUM>",
+      "position" : 0
+    },
+    {
+      "token" : "schiff",
+      "start_offset" : 0,
+      "end_offset" : 16,
+      "type" : "<ALPHANUM>",
+      "position" : 0
+    }
+  ]
+}
+--------------------------------------------------
+/////////////////////
+
+[[analysis-dict-decomp-tokenfilter-configure-parms]]
+==== Configurable parameters
+
+`word_list`::
++
+--
+(Required+++*+++, array of strings)
+A list of subwords to find in the token stream. If found, the subword is
+included in the token output.
+
+Either this or the `word_list_path` parameter is required.
+--
+
+`word_list_path`::
++
+--
+(Required+++*+++, string)
+Path to a file containing a list of subwords to find in the token stream. If
```

Comment:
```suggestion
Path to a file that contains a list of subwords to find in the token stream. If
```
Getting into picky grammar-tics that are more prescriptive than truly necessary.